### ISSUE TITLE
[Rootful in UserNS] Regression: systemd rootless attempted when cgroup v1 in use

### ISSUE THREAD

**Issue by danopia** [opencontainers/runc#2724]
Hello, I've noticed a runc regression by using Podman on a Chromebook's 'Linux beta' environment, which is furnished via LXC. This is the array of conditions at play:

* Running podman/runc as root
* Apparently within a userns (presumably because of LXC)
* Systemd is running
* cgroup v1 *is* mounted, cgroup v2 *not* mounted

Given all this, latest runc fails:

```
> git checkout v1.0.0-rc92
> make
> make install
> sudo podman run --rm busybox echo hi world
Error: OCI runtime error: cgroup v2 not enabled on this host, can't use systemd (rootless) as cgroups manager
```

Debian ships with this ancient runc which works:

```
> runc --version
runc version 1.0.0~rc6+dfsg1
commit: 1.0.0~rc6+dfsg1-3
spec: 1.0.1
> sudo podman run --rm busybox echo hi world
hi world
```

I've written a simple git-bisect script to find the exact cause, which pointed at https://github.com/opencontainers/runc/commit/bf15cc99b1e7b486dfbb394992fcc4b38157d56e .

Seems like systemd-rootless shouldn't be enabled unless cgroupv2 is enabled.

Initial investigative work happened here: https://github.com/containers/podman/issues/8881

**Comment by kolyshkin** [comment]
AFAIK systemd cgroups driver is off by default (unless explicitly enabled via `--systemd-cgroup`).

**Comment by kolyshkin** [comment]
And what the error message is trying to say is "you can't use rootless + systemd cgroup driver with cgroup v1".

**Comment by kolyshkin** [comment]
@AkihiroSuda PTAL

**Comment by danopia** [comment]
podman only seems to reference that flag the `build` codepath while I'm running `run`:

```
> git grep systemd-cgroup
cmd/podman/images/build.go:             runtimeFlags = append(runtimeFlags, "--systemd-cgroup")
vendor/github.com/docker/docker/api/swagger.yaml:            runtimeArgs: ["--debug", "--systemd-cgroup=false"]
vendor/github.com/docker/docker/api/swagger.yaml:        example: ["--debug", "--systemd-cgroup=false"]
```

I suppose it's possible it's coming from somewhere else? If it is then presumably the old runcs only work because they're just ignoring that flag?

**Comment by AkihiroSuda** [comment]
Fix: https://github.com/opencontainers/runc/pull/2725

**Comment by danopia** [comment]
I've confirmed that the fix in #2725 resolves my issue. Thanks!
